### PR TITLE
Add missing CoreData tests and some improvements 🛢

### PR DESCRIPTION
**Pull request by p4checo**
## Changes

- Added missing CoreData UT's.

- Did some refinements to `CoreDataStack` API's.

- Did some file reorganization and code cleanups.

### CODE REVIEW

File: Sources/Persistence/CoreData/NSManagedObjectContext+CoreDataStack.swift [hunk 1]
```diff
@@ -36,54 +47,71 @@ public extension NSManagedObjectContext {
...
     }
 
-    func isSQLiteStoreBased() -> Bool {
-        switch topLevelPersistentStoreCoordinator().firstStoreType() {
-        case .sqLite: return true
-        default: return false
+    var isSQLiteStoreBased: Bool {
+
+        switch topLevelPersistentStoreCoordinator?.firstStoreType {
+        case .sqLite?:
```
Comment (filipe-lemos) [inline Sources/Persistence/CoreData/NSManagedObjectContext+CoreDataStack.swift]: As discussed, maybe rethink capitalization in `.sqLite`

Comment (belosoGrocerKey) [reply Sources/Persistence/CoreData/NSManagedObjectContext+CoreDataStack.swift]: sqlite?

Comment (beloso) [reply Sources/Persistence/CoreData/NSManagedObjectContext+CoreDataStack.swift]: `sqlite`?

Comment (p4checo) [reply Sources/Persistence/CoreData/NSManagedObjectContext+CoreDataStack.swift]: either that or `SQLite`, any preference? 🚲

**Pull request merged**